great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -4721,8 +4721,83 @@ def _initialize_usage_statistics(
             data_context=self,
             data_context_id=self._data_context_id,
             usage_statistics_url=usage_statistics_config.usage_statistics_url,
+            oss_id=self._get_oss_id(),
         )
 
+    @classmethod
+    def _get_oss_id(cls) -> uuid.UUID | None:
+        """
+        Retrieves a user's `oss_id` from disk ($HOME/.great_expectations/great_expectations.conf).
+
+        If no such value is present, a new UUID is generated and written to disk for subsequent usage.
+        If there is an error when reading from / writing to disk, we default to a NoneType.
+        """
+        config = configparser.ConfigParser()
+
+        if not cls._ROOT_CONF_FILE.exists():
+            success = cls._scaffold_root_conf()
+            if not success:
+                return None
+            return cls._set_oss_id(config)
+
+        try:
+            config.read(cls._ROOT_CONF_FILE)
+        except OSError as e:
```

Comment:
I could have been more granular but opting for safety with the higher level `OSError` - any qualms?